includes/blocks/course-theme/class-prev-next-lesson.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * File containing the Prev_Next_Lesson class.
+ *
+ * @package sensei
+ * @since
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+use \Sensei\Blocks\Course_Theme\Prev_Lesson;
+use \Sensei\Blocks\Course_Theme\Next_Lesson;
+
+/**
+ * Class Prev_Next_Lesson is responsible for rendering the '< Prev Lesson | Next Lesson >' blocks.
+ */
+class Prev_Next_Lesson {
+
+	/**
+	 * Reference to the previous lesson button block.
+	 *
+	 * @var Sensei_CT_Prev_Lesson_Block
+	 */
+	private $prev = null;
+
+	/**
+	 * Reference to the previous lesson button block.
+	 *
+	 * @var Sensei_CT_Next_Lesson_Block
+	 */
+	private $next = null;
+
+	/**
+	 * Prev_Next_Lesson constructor.
+	 *
+	 * @param Prev_Lesson $prev The previous lesson block.
+	 * @param Next_Lesson $next The next lesson block.
+	 */
+	public function __construct( Prev_Lesson $prev, Next_Lesson $next ) {
+		$this->prev = $prev;
+		$this->next = $next;
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-prev-next-lesson',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render() : string {
+		$prev = $this->prev->render();
+		$next = $this->next->render();
+		return "<div  class='sensei-course-theme-prev-next-lesson-container'>$prev $next</div>";
```

Comment:
My thinking was to separate code into smaller pieces so it is easier when we implement them as blocks in traditional sense. But ultimately the objective was to create blocks that could be used in the template file and rendered with `do_blocks()`.

In this particular case I would have to also implement the JS part of it so it works correctly with prev-lesson & next-lesson as nested blocks here.

Reply:
For parent/nested blocks, it would be a [little different](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/) with InnerBlocks.

I think the problem here is when we call the method that is the [`render_callback`](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/creating-dynamic-blocks/) of the other blocks. If we want to separate the methods, it wouldn't be a problem, but calling the methods that are the `render_callback` of the other blocks probably isn't a good practice. Basically, the `render_callback` doesn't represent exactly the block render, but a "tweak" in the block, because it can also receive the block attributes and the content (saved in the post_content through the JS).

Reply:
OK, I think I started understand what you're saying now. So the `render_callback` is not intended for manual use, but is used  by WP internally instead. The way I use it here now is likely to cause unexpected behavior in the future.

I will make sure to remove that in next iteration. 👍

Reply:
Addressed here https://github.com/Automattic/sensei/pull/4432/files#diff-67dd29b332b14fb28fe96b925474ff7c7b5db41d6d70034a58c0004c0c034f61

Now it renders inner blocks contents instead. The inner blocks should be added manually on the code editor like this.
```
<!-- wp:sensei-lms/course-theme-prev-next-lesson -->
<!-- wp:sensei-lms/course-theme-prev-lesson {"inContainer":true} /-->
<!-- wp:sensei-lms/course-theme-next-lesson {"inContainer":true} /-->
<!-- /wp:sensei-lms/course-theme-prev-next-lesson -->
```